### PR TITLE
Added BCryptPasswordEncoder

### PR DESCRIPTION
**Pull request by maxedel**

### CODE REVIEW

File: flix-core/src/main/java/com/epam/mentoring/flixcore/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -22,13 +24,25 @@
...
     @Override
-    public void createUser(User user, Role role) {
-        UserRole userRole = userRoleRepository.findByRole(role);
-        user.setUserRoles(new HashSet<>(Collections.singletonList(userRole)));
+    public void saveUser(User user) {
+        System.out.println("1user = " + user);
+        user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
+        user.setActive(1);
+        UserRole userRole = userRoleRepository.findByRole(Role.ADMIN);
+        user.setUserRoles(new HashSet<>(Arrays.asList(userRole)));
```
Comment (ManWithNoNamePL) [inline flix-core/src/main/java/com/epam/mentoring/flixcore/service/impl/UserServiceImpl.java]: Set.of(userRole)

File: flix-core/src/main/java/com/epam/mentoring/flixcore/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -22,13 +24,25 @@
...
+        System.out.println("2user = " + user);
```
Comment (ManWithNoNamePL) [inline flix-core/src/main/java/com/epam/mentoring/flixcore/service/impl/UserServiceImpl.java]: dont do System.out, wanna print something use log

File: flix-core/src/main/java/com/epam/mentoring/flixcore/controller/LoginController.java [hunk 3]
```diff
@@ -37,18 +35,18 @@ public ModelAndView registration() {
         return modelAndView;
     }
 
-    @RequestMapping(value = "/registration", method = RequestMethod.POST)
+    @PostMapping(value = "/registration")
     public ModelAndView createNewUser(@Valid User user, BindingResult bindingResult) {
```
Comment (ManWithNoNamePL) [inline flix-core/src/main/java/com/epam/mentoring/flixcore/controller/LoginController.java]: small hint here - BindingResult MUST be just after @Valid bean - you did it correct, but maybe accidentally :)
Keep this in mind coz if its not next it just wont work and wont tell you why. Very nasty bug.

File: flix-core/src/main/java/com/epam/mentoring/flixcore/service/impl/UserServiceImpl.java (hunk 2 above)
Comment (maxedel) [reply flix-core/src/main/java/com/epam/mentoring/flixcore/service/impl/UserServiceImpl.java]: Removed

**Pull request merged**